packages/calendar/src/Calendar.ts

```diff
@@ -0,0 +1,449 @@
+            `${todayTitle}${currentDayTitle}${selectedTitle}`
+        );
+
+        return html`
+            <td
+                role="gridcell"
+                class="spectrum-Calendar-tableCell"
+                title=${title}
+                tabindex=${ifDefined(!isOutsideMonth ? '-1' : undefined)}
+                aria-disabled=${isOutsideMonth || this.disabled}
+                aria-selected=${isSelected}
+            >
+                <span
+                    role="presentation"
+                    class="spectrum-Calendar-date ${classMap(dayClasses)}"
+                    @click=${() => this.handleDayClick(calendarDate)}
+                >
+                    ${this._formatNumber(calendarDate.day)}
+                </span>
+            </td>
+        `;
+    }
+
+    public handlePreviousMonth(): void {
+        this._currentDate = startOfMonth(this._currentDate).subtract({
+            months: 1,
+        });
+    }
+
+    public handleNextMonth(): void {
+        this._currentDate = startOfMonth(this._currentDate).add({ months: 1 });
+    }
+
+    public handleDayClick(calendarDate: CalendarDate): void {
+        this.selectedDate = calendarDate.toDate(this._timeZone);
+        this._setCurrentCalendarDate();
+
+        this.dispatchEvent(
+            new CustomEvent('change', {
+                bubbles: true,
+                cancelable: true,
+                composed: true,
+                detail: this.selectedDate,
+            })
+        );
+    }
+
+    private _getWeeksInCurrentMonth(): number {
+        return getWeeksInMonth(this._currentDate, this._locale);
+    }
+
+    /**
+     * Returns data for the days of the week, starting with the first day of the week according to the defined locale
+     * (Sunday, Monday, etc.)
+     */
+    private _getWeekdays(): CalendarWeekday[] {
+        return [...new Array(daysInWeek).keys()].map((dayIndex) => {
+            const weekStart = startOfWeek(this._currentDate, this._locale);
+            const date = weekStart.add({ days: dayIndex });
+
+            return {
+                narrow: this._formatDate(date, { weekday: 'narrow' }),
+                long: this._capitalFirstLetter(
+                    this._formatDate(date, { weekday: 'long' })
+                ),
+            };
+        });
+    }
+
+    private _setLocale(): void {
+        this._locale = this._languageResolver.language;
+    }
+
+    private _setInitialCalendarDate(): void {
+        this._currentDate = this.today;
+    }
+
+    private _setCurrentCalendarDate(): void {
+        if (this.selectedDate) {
+            this.selectedDate = new Date(this.selectedDate);
+
+            if (!this._isValidDate(this.selectedDate)) {
+                this.selectedDate = undefined;
+            } else {
+                this._currentDate = this._toCalendarDate(this.selectedDate);
+            }
+        }
+    }
+
+    private _setMinCalendarDate(): void {
+        if (this.min) {
```

Comment:
You can use the early return pattern here which is good in catching early errors into your methods